File: sponge-jruby/examples/script/rb/rules_sync_async.rb
# Sponge Knowledge base
# Using rules - synchronous and asynchronous

java_import org.openksavi.sponge.test.util.CorrelationEventsLog

def onInit
    # Variables for assertions only
    $correlationEventsLog = CorrelationEventsLog.new
    $EPS.setVariable("correlationEventsLog", $correlationEventsLog)
end

class RuleFFF < Rule
    def onConfigure
        self.events = ["e1", "e2", "e3 :first"]
        self.synchronous = true
    end
    def onRun(event)
        self.logger.debug("Running rule for event: {}", event.name)
        $correlationEventsLog.addEvents("RuleFFF", self)
    end
end

class RuleFFL < Rule
    def onConfigure
        self.events = ["e1", "e2", "e3 :last"]
        self.duration = Duration.ofMillis(500)
        self.synchronous = false
    end
    def onRun(event)
        self.logger.debug("Running rule for event: {}", event.name)
        $correlationEventsLog.addEvents("RuleFFL", self)
    end
end

def onStartup
    $EPS.event("e1").set("label", "1").send()
    $EPS.event("e2").set("label", "2").send()
    $EPS.event("e2").set("label", "3").send()
    $EPS.event("e2").set("label", "4").send()
    $EPS.event("e3").set("label", "5").send()
    $EPS.event("e3").set("label", "6").send()
    $EPS.event("e3").set("label", "7").send()
end

File: sponge-jython/examples/script/py/rules_sync_async.py
"""
Sponge Knowledge base
Using rules - synchronous and asynchronous
"""

from org.openksavi.sponge.test.util import CorrelationEventsLog

def onInit():
    global correlationEventsLog

    # Variables for assertions only
    correlationEventsLog = CorrelationEventsLog()
    sponge.setVariable("correlationEventsLog", correlationEventsLog)

class RuleFFF(Rule):
    def onConfigure(self):
        self.withEvents(["e1", "e2", "e3 :first"]).withSynchronous(True)
    def onRun(self, event):
        self.logger.debug("Running rule for event: {}", event.name)
        global correlationEventsLog
        correlationEventsLog.addEvents("RuleFFF", self)

class RuleFFL(Rule):
    def onConfigure(self):
        global defaultDuration
        self.withEvents(["e1", "e2", "e3 :last"]).withDuration(Duration.ofSeconds(2)).withSynchronous(False)
    def onRun(self, event):
        self.logger.debug("Running rule for event: {}", event.name)
        global correlationEventsLog
        correlationEventsLog.addEvents("RuleFFL", self)

def onStartup():
    sponge.event("e1").set("label", "1").send()
    sponge.event("e2").set("label", "2").send()
    sponge.event("e2").set("label", "3").send()
    sponge.event("e2").set("label", "4").send()
    sponge.event("e3").set("label", "5").send()
    sponge.event("e3").set("label", "6").send()
    sponge.event("e3").set("label", "7").send()
